Connectors/Tekla/Speckle.Connector.TeklaShared/Bindings/TeklaSelectionBinding.cs

```diff
@@ -9,29 +8,34 @@ public class TeklaSelectionBinding : ISelectionBinding
 {
   private const string SELECTION_EVENT = "setSelection";
   private readonly object _selectionEventHandlerLock = new object();
+  private readonly IAppIdleManager _idleManager;
+  private readonly Events _events;
   private readonly Tekla.Structures.Model.UI.ModelObjectSelector _selector;
-  private readonly IEventAggregator _eventAggregator;
 
   public string Name => "selectionBinding";
   public IBrowserBridge Parent { get; }
 
   public TeklaSelectionBinding(
+    IAppIdleManager idleManager,
     IBrowserBridge parent,
-    Tekla.Structures.Model.UI.ModelObjectSelector selector,
-    IEventAggregator eventAggregator
+    Events events,
+    Tekla.Structures.Model.UI.ModelObjectSelector selector
   )
   {
+    _idleManager = idleManager;
     Parent = parent;
     _selector = selector;
-    _eventAggregator = eventAggregator;
+    _events = events;
 
-    eventAggregator.GetEvent<SelectionChangeEvent>().Subscribe(OnSelectionChangeEvent);
+    _events.SelectionChange += OnSelectionChangeEvent;
+    _events.Register();
   }
 
-  private void OnSelectionChangeEvent(object _)
+  private void OnSelectionChangeEvent()
   {
     lock (_selectionEventHandlerLock)
     {
+      _idleManager.SubscribeToIdle(nameof(TeklaSelectionBinding), UpdateSelection);
       UpdateSelection();
```

Comment:
Did you intend to call queue `UpdateSelection` on the idle manager, and then immediately call it anyway? Surely one or the other is correct.

Reply:
definetely not remember why, @dogukankaratas can have better context about why we did like this